### PR TITLE
MS Graph Groups and Users (Independent Publisher)

### PR DESCRIPTION
**Pull request by jagjani03**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [X] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [X] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
image

- [X] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
image

- [X] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by v-EgorKozhadei** [comment]
Hi @jagjani03,
Congratulations on your new submission! 

You need to sign the CLA before we can start the review process.
After this step is done our Certification Engineers will review your submission within 1-2 weeks.

Please let us know if you have any questions.

Thank you very much for working with us.

File: independent-publisher-connectors/MS Graph Groups and Users/readme.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+### Get Member License Details
+Retrieve group member(user)'s license details.
+
+### Get Group Properties
+Retrieve properties and relationships of an AAD group.
+
+### Get Member Groups
+Group memberships for a user (member). This operation will retrieve list of AAD groups the member belongs to.
+
+## API Documentation
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/MS Graph Groups and Users/readme.md:44]: @jagjani03 please format the readme.md file according to [the template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md?plain=1), all of the required headers should be present.

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,1151 @@
...
+                },
+                "visibility": {
+                  "type": "string",
+                  "description": "visibility",
+                  "title": "Visibility"
+                },
+                "onPremisesProvisioningErrors": {
+                  "type": "array",
+                  "items": {
+                    "description": "onPremisesProvisioningErrors",
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json:1001]: @jagjani03 please look through all the descriptions and format them according to the guidelines provided in previous comments.

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @jagjani03! Please check comments after the review, thank you!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,1151 @@
...
+        "description": "Retrieve all users in the organization (AAD Tenant)",
+        "operationId": "ListUsers",
+        "parameters": []
+      }
+    },
+    "/v1.0/groups": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "default",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,1151 @@
...
+    "https"
+  ],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "paths": {
+    "/v1.0/users": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "default",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,1151 @@
...
+            "schema": {
+              "type": "object",
+              "properties": {
+                "@odata.context": {
+                  "type": "string",
+                  "description": "@odata.context"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1151 @@
...
+      "get": {
+        "responses": {
+          "200": {
+            "description": "default",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "@odata.context": {
+                  "type": "string",
+                  "description": "@odata.context",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1151 @@
...
+                      },
+                      "createdDateTime": {
+                        "type": "string",
+                        "description": "Date and time of group creation",
+                        "title": "Created Date Time"
+                      },
+                      "creationOptions": {
+                        "type": "array",
+                        "items": {},
+                        "description": "creationOptions"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,1151 @@
...
+                        ]
+                      },
+                      "mailNickname": {
+                        "type": "string",
+                        "description": "Group's email nick name",
+                        "title": "Mail Nick Name"
+                      },
+                      "membershipRule": {
+                        "type": "string",
+                        "description": "membershipRule",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Membership Rule"
+                      },
+                      "membershipRuleProcessingState": {
+                        "type": "string",
+                        "description": "membershipRuleProcessingState",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Membership Rule Processing State"
+                      },
+                      "onPremisesDomainName": {
+                        "type": "string",
+                        "description": "onPremisesDomainName",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "OnPremises Domain Name"
+                      },
+                      "onPremisesLastSyncDateTime": {
+                        "type": "string",
+                        "description": "onPremisesLastSyncDateTime",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "OnPremises Last Sync Date Time"
+                      },
+                      "onPremisesNetBiosName": {
+                        "type": "string",
+                        "description": "onPremisesNetBiosName",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "OnPremises NetBios Name"
+                      },
+                      "onPremisesSamAccountName": {
+                        "type": "string",
+                        "description": "onPremisesSamAccountName",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "OnPremises Sam Account Name"
+                      },
+                      "onPremisesSecurityIdentifier": {
+                        "type": "string",
+                        "description": "onPremisesSecurityIdentifier",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 15]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "OnPremises Security Identifier"
+                      },
+                      "onPremisesSyncEnabled": {
+                        "type": "string",
+                        "description": "onPremisesSyncEnabled",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 16]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "OnPremises Sync Enabled"
+                      },
+                      "preferredDataLocation": {
+                        "type": "string",
+                        "description": "preferredDataLocation",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 17]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Preferred Data Location"
+                      },
+                      "preferredLanguage": {
+                        "type": "string",
+                        "description": "preferredLanguage",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 18]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Preferred Language"
+                      },
+                      "proxyAddresses": {
+                        "type": "array",
+                        "items": {
+                          "type": "string"
+                        },
+                        "description": "proxyAddresses"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 19]
```diff
@@ -0,0 +1,1151 @@
...
+                      },
+                      "renewedDateTime": {
+                        "type": "string",
+                        "description": "renewedDateTime",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 20]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Renewed Date Time"
+                      },
+                      "resourceBehaviorOptions": {
+                        "type": "array",
+                        "items": {},
+                        "description": "resourceBehaviorOptions"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 21]
```diff
@@ -0,0 +1,1151 @@
...
+                      },
+                      "resourceProvisioningOptions": {
+                        "type": "array",
+                        "items": {
+                          "type": "string"
+                        },
+                        "description": "resourceProvisioningOptions"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 22]
```diff
@@ -0,0 +1,1151 @@
...
+                        "description": "securityEnabled",
+                        "title": "Security Enabled",
+                        "enum": [
+                          true,
+                          false
+                        ]
+                      },
+                      "securityIdentifier": {
+                        "type": "string",
+                        "description": "securityIdentifier",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 23]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Security Identifier"
+                      },
+                      "theme": {
+                        "type": "string",
+                        "description": "theme",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 24]
```diff
@@ -0,0 +1,1151 @@
...
+                      },
+                      "visibility": {
+                        "type": "string",
+                        "description": "visibility",
+                        "title": "Visibility"
+                      },
+                      "onPremisesProvisioningErrors": {
+                        "type": "array",
+                        "items": {},
+                        "description": "onPremisesProvisioningErrors"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 25]
```diff
@@ -0,0 +1,1151 @@
...
+                        "title": "Security Identifier"
+                      },
+                      "theme": {
+                        "type": "string",
+                        "description": "theme",
+                        "title": "Theme"
+                      },
+                      "visibility": {
+                        "type": "string",
+                        "description": "visibility",
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 26]
```diff
@@ -0,0 +1,1151 @@
...
+                            "title": "Suspended"
+                          },
+                          "warning": {
+                            "type": "integer",
+                            "format": "int32",
+                            "description": "Units in warning",
+                            "title": "Warning"
+                          }
+                        },
+                        "description": "prepaidUnits"
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/readme.md (hunk 1 above)
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/readme.md:46]: @v-EgorKozhadei readme.md file has all the required headers per template provided. Please review. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json (hunk 2 above)
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei description has been updated. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 27]
```diff
@@ -0,0 +1,1151 @@
...
+            "description": "Object ID of the AAD group",
+            "x-ms-summary": "Object ID of the AAD group",
+            "x-ms-url-encoding": "single"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "jobTitle ne null",
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json:661]: @jagjani03 can you please check if this text is correct? Shouldn't it be `jobTitle is not null`?

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 28]
```diff
@@ -0,0 +1,1151 @@
...
+                      },
+                      "skuId": {
+                        "type": "string",
+                        "description": "skuId",
+                        "title": "Sku ID"
+                      },
+                      "skuPartNumber": {
+                        "type": "string",
+                        "description": "Name of the sku",
+                        "title": "Sku Part Number"
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json:724]: @jagjani03 if you use abbreviations, please make sure to capitalize them to make it clear.

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @jagjani03! Thank you for all the fixes! Please check a few comments, we'll proceed to the next step after a minor fix.

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json (hunk 27 above)
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json:661]: @v-EgorKozhadei This is correct. It's oData syntax. Thanks!

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json (hunk 28 above)
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: @v-EgorKozhadei This has been completed. Thanks!

**Comment by v-EgorKozhadei** [comment]
[[certify-connector]]

**Comment by v-EgorKozhadei** [comment]
Hello @jagjani03, 

I hope you are doing well. 

Congratulations, your connector is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.  

Please let us know if you have any questions. 

Thank you very much for working with us.

File: independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json [hunk 29]
```diff
@@ -0,0 +1,1151 @@
...
+  "definitions": {},
+  "parameters": {},
+  "responses": {},
+  "securityDefinitions": {
+    "oauth2_auth": {
+      "type": "oauth2",
+      "flow": "accessCode",
+      "authorizationUrl": "https://login.windows.net/common/oauth2/authorize",
+      "tokenUrl": "https://login.windows.net/common/oauth2/authorize",
+      "scopes": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json:1142]: @jagjani03, can you please specify the scopes here? Thank you very much.

Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiDefinition.swagger.json]: Hi @Amjed-Ayoub scopes have been updated. Thank you!

File: independent-publisher-connectors/MS Graph Groups and Users/apiProperties.json [hunk 30]
```diff
@@ -0,0 +1,48 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "aad",
+          "clientId": "{{aad_application_id}}",
+          "scopes": [],
```
[the remark this answers is not in the document]
Comment (jagjani03) [reply independent-publisher-connectors/MS Graph Groups and Users/apiProperties.json]: Hi @Amjed-Ayoub scopes have been updated. Thank you!

**Comment by sriyen-msft** [comment]
@jagjani03 We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments which typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps. Thanks.

**Comment by jagjani03** [comment]
@sriyen-msft Thank you!

**Comment by v-EgorKozhadei** [comment]
Hello @jagjani03, 

I hope you are doing well. 

Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program. 

Thank you very much for working with us.

**Review by v-EgorKozhadei — APPROVED** [review]

**Pull request merged**